### PR TITLE
[junit] Use uuidV4 for filename

### PR DESCRIPTION
**Pull request by rodrigo-roca**
### What and why?

If we send the normal filename it can contain some characters that would cause 400 errors at the intake. Now we create a [uuidv4](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)) for the filename and send the original filename somewhere else.

### How?

Use uuid package and send filename somewhere else as `_dd.report_name`

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/junit/api.ts
Comment (AdrianLC) [inline src/commands/junit/api.ts:60]: We want to apply this also to the filename in sarif, don't we?

Comment (rodrigo-roca) [reply src/commands/junit/api.ts:60]: Is it the same as junit that EvP handles the uniqueness of the upload? I am not very knowledgeable about that part

Comment (AdrianLC) [reply src/commands/junit/api.ts:60]: yes the upload to s3 is common

**Review by AdrianLC — APPROVED** [review]
LGTM but let's wait until the backend change is deployed

**Review by juan-fernandez — APPROVED** [review]
LGTM, but we need to add the new dependencies to https://github.com/DataDog/datadog-ci/blob/master/LICENSE-3rdparty.csv.

Also, let's wait for the approval from someone else from datadog-ci team, since this is adding a new dependency and it affects other commands

**Review by duncanista — APPROVED** [review]
LGTM – on the dependencies side!

**Pull request merged**
